src/core/lib.lua

```diff
@@ -386,12 +353,64 @@ if ffi.abi("be") then
    function htonl(b) return b end
    function htons(b) return b end
 else
-   function htonl(b) return bit.bswap(b) end
-   function htons(b) return bit.rshift(bit.bswap(b), 16) end
+   function htonl(b) return bswap(b) end
+   function htons(b) return rshift(bswap(b), 16) end
 end
 ntohl = htonl
 ntohs = htons
 
+-- The fact that BitOps return signed integers is irritating, to say
+-- the least.  One counter-intuitive consequence is that, for example,
+--
+--   bit.bswap(bit.bswap(0x80000000)) == 0x80000000
```

Comment:
To avoid this problem, constant numbers should be normalized using `bit.tobit`. Example:

```lua
> if bit.bswap(bit.bswap(0x80000000)) == 0x80000000 then print("true") else print("false") end
false
> if bit.bswap(bit.bswap(0x80000000)) == bit.tobit(0x80000000) then print("true") else print("false") end
true
```

According to LuaJIT docs "this function is usually not needed since all bit operations already normalize all of their input arguments" [1], but in the case of constants is necessary to call it explictly.

[1] http://bitop.luajit.org/api.html